File: trees/java/Arbol.java

public class Arbol {
  private String key;
  private Arbol left;
  private Arbol right;

  public Arbol(String key, Arbol left, Arbol right) {
    this.key = key;
    this.left = left;
    this.right = right;
  }

  public String getKey() {
    return this.key;
  }

  public void setKey(String key) {
    this.key = key;
  }

  public Arbol getLeft() {
    return this.left;
  }

  public void setLeft(Arbol left) {
    this.left = left;
  }

  public Arbol getRight() {
    return this.right;
  }

  public void setRight(Arbol right) {
    this.right = right;
  }
}

File: trees/typescript/Arbol.ts
type ArbolType<T> = Arbol<T>
type ArbolMainType<T> = ArbolType<T> | null


export class Arbol<T> {
  private _key: T
  private _left: ArbolMainType<T>
  private _right: ArbolMainType<T>

  constructor(_key: T, _left: ArbolMainType<T>, _right: ArbolMainType<T>) {
    this._key = _key
    this._left = _left
    this._right = _right
  }

  get key(): T {
    return this._key
  }

  set key(_key: T) {
    this._key = _key
  }

  get left(): ArbolMainType<T> {
    return this._left
  }

  set left(_left: ArbolMainType<T>) {
    this._left = _left
  }

  get right(): ArbolMainType<T> {
    return this._right
  }

  set right(_right: ArbolMainType<T>) {
    this._right = _right
  }
}
